test/monitoring/components_metrics_test.go

```diff
@@ -24,47 +24,77 @@ import (
 	"testing"
 	"time"
 
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+
 	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
 )
 
+//nolint:funlen
 func testComponentsPrometheusMetrics(t *testing.T, v1api v1.API) {
 	testCases := []struct {
 		componentName string
 		query         string
+		platforms     map[string]interface{}
 	}{
 		{
 			componentName: "kube-apiserver",
 			query:         "apiserver_request_total",
+			platforms:     map[string]interface{}{"*": nil},
```

Comment:
This seems redundant. If the check should be done on all platforms, can we simply skip the field?

Reply:
This is not checking on all platforms, it is a special string which means the platform checker function always returns true.

Reply:
Why can't you check if `platforms` is nil instead of specific value?

Reply:
Because there could be cases where the test author misses it and adds tests with bugs. So it is rather better to be explicit here. And since people are familiar with the usage of `*` the way it is done in kubernetes RBAC, I thought this was the most obvious thing to use.

Reply:
>Because there could be cases where the test author misses it and adds tests with bugs.

Can you elaborate on that please? I don't understand it.

> And since people are familiar with the usage of * the way it is done in kubernetes RBAC, I thought this was the most obvious thing to use.

I understand the rationale, but `foo: "*"` in YAML is simpler than `map[string]interface{}{"*": nil},` and also, we don't deal with YAMLs here, but with Go structs.

I'd say the argument for just skipping it here is to ["make the zero value useful"](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=6m25s), even though it's not 100% accurate.

Reply:
done by changing the mechanism itself. PTAL.